commands/base_command.go

```diff
@@ -213,8 +213,18 @@ func (c *BaseCommand) GetUsername() (string, error) {
 }
 
 // GetDeployServiceURL returns the deploy service URL
-func (c *BaseCommand) GetDeployServiceURL() (string, error) {
-	deployServiceURL := configuration.GetTargetURL()
+func (c *BaseCommand) GetDeployServiceURL(args []string) (string, error) {
+	var deployServiceURL string
+
+	envVarDeployServiceURL := configuration.GetTargetURL()
+
+	deployServiceURL = envVarDeployServiceURL
+
+	optionDeployServiceURL := GetOptionValue(args, deployServiceURLOpt)
```

Comment:
Extract this and the above three lines in a method.
In it:
1. Check if the envVarDeployServiceURL != "" -> return it
2. Otherwise return GetOptionValue(args, deployServiceURLOpt)
3. Assign the result to the optionDeployServiceURL

Reply:
Actually the proposed logic is not correct.
Value coming from option "-u" should override the value from env var DEPLOY_SERVICE_URL.

However, it is extracted in a new function